src/cloudai/configurator/trajectory.py

```
@@ -0,0 +1,405 @@
+    def find(self, action: Mapping[str, Any], **identity_values: object) -> TrajectoryEntry | None:
+        identity_components = self._construct_components(
+            tuple(component_type for component_type in self._component_types if component_type in self._identity),
+            identity_values,
+            "trajectory identity values",
+        )
+        identity = self._identity_for(identity_components)
+        frozen_action = _freeze(action)
+        for entry in self._entries:
+            result = entry.get(TrialResult)
+            if result is None:
+                raise ValueError(f"trajectory entry at step {entry.step} is missing TrialResult")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)